### DISCUSSION TITLE
Accessing response options (options as in "new Response(body, options)") from the page comp

### DISCUSSION THREAD

**Discussion by pagebanc** [remix-run/remix#3892] [Q&A] [answered]
Hi,
Consider a loader function (this is from https://remix.run/docs/en/v1/api/remix#json):
```
export const loader: LoaderFunction = async () => {
  return json(
    { not: "coffee" },
    {
      status: 418,
      headers: {
        "Cache-Control": "no-store",
      },
    }
  );
};
```

There are two parts here, data ("coffee" in this example) and options (status code, etc.).
In the page component, the **data** can be retrieved thru useLoaderData:
```
const { not } = useLoaderData()
```
But is there a way to access the **options part** from the page component?

Thank you.

**Comment by kiliman** [comment, marked as answer]
Remix only returns the data part via `useLoaderData`. You don't have access to the Response status or headers. If you need that information, you should return it as part of your data.

**Reply by pagebanc** [reply]
Ok, thanks!
